v2/src/log_sink_etw.c

```diff
@@ -0,0 +1,497 @@
+{
+    TraceLoggingHProvider const _tlgProv = g_my_component_provider;
+    if (trace_level < _tlgProv->LevelPlus1)
+    {
+        // have a stack allocated buffer where we construct the event metadata and
+        uint8_t _tlgEvent[MAX_SELF_DESCRIBED_EVENT_WITH_METADATA];
+        bool add_properties = true;
+
+        // compute event metadata size
+        // event_name_length includes NULL terminator
+        /* Codes_SRS_LOG_SINK_ETW_01_042: [ log_sink_etw.log_sink_log shall compute the metadata size for the self described event metadata as follows: ]*/
+        /* Codes_SRS_LOG_SINK_ETW_01_043: [ Length of the event name (determined at compile time, excluding zero terminator) + 1. ]*/
+        uint32_t i;
+
+        // alias to the event bytes
+        SELF_DESCRIBED_EVENT* self_described_event = (SELF_DESCRIBED_EVENT*)_tlgEvent;
+
+        uint8_t* pos = &self_described_event->metadata[0];
+
+        // at this point we filled all the event information, now we need to provide the metadata bytes
+        // first one is the event name, followed by metadata for all the fields
+        // copy event name 
+        (void)memcpy(pos, event_name, event_name_length); pos += event_name_length;
+
+        // copy the field metadata (name and in type)
+
+        /* Codes_SRS_LOG_SINK_ETW_01_034: [ log_sink_etw.log_sink_log shall fill the event metadata: ]*/
+
+        /* Codes_SRS_LOG_SINK_ETW_01_035: [ The string content (as field name, excluding zero terminator), followed by one byte with the value TlgInANSISTRING. ]*/
+        /* Codes_SRS_LOG_SINK_ETW_01_044: [ Length of the content field name (determined at compile time, excluding zero terminator) + 1. ]*/
+        (void)memcpy(pos, "content", sizeof("content")); pos += sizeof("content");
+        *pos = TlgInANSISTRING;  pos++;
+
+        /* Codes_SRS_LOG_SINK_ETW_01_036: [ The string file (as field name, excluding zero terminator), followed by one byte with the value TlgInANSISTRING. ]*/
+        /* Codes_SRS_LOG_SINK_ETW_01_045: [ Length of the file field name (determined at compile time, excluding zero terminator) + 1. ]*/
+        (void)memcpy(pos, "file", sizeof("file")); pos += sizeof("file");
+        *pos = TlgInANSISTRING;  pos++;
+
+        /* Codes_SRS_LOG_SINK_ETW_01_037: [ The string func (as field name, excluding zero terminator), followed by one byte with the value TlgInANSISTRING. ]*/
+        /* Codes_SRS_LOG_SINK_ETW_01_046: [ Length of the func field name (determined at compile time, excluding zero terminator) + 1. ]*/
+        (void)memcpy(pos, "func", sizeof("func")); pos += sizeof("func");
+        *pos = TlgInANSISTRING;  pos++;
+
+        /* Codes_SRS_LOG_SINK_ETW_01_038: [ The string line (as field name, excluding zero terminator), followed by one byte with the value TlgInINT32. ]*/
+        /* Codes_SRS_LOG_SINK_ETW_01_047: [ Length of the line field name (determined at compile time, excluding zero terminator) + 1. ]*/
```

Comment:
>Length [](http://example.com/codeflow?start=44&length=6)

If you say "size of" then the +1 from the end of the spec is not needed.

Reply:
Changed